### PR TITLE
Small performance improvement in h3 library

### PR DESCRIPTION
**Pull request by iverase**
In https://github.com/uber/h3/pull/852, it is claimed that changing some FDIV's into FMUL's leads to performance improvements. this is due to modern CPUs are faster doing multiplications that divisions. I follow the same idea in the Java H3 library and it leads to improvements in micro-benachmarks:

before:
```
Benchmark                Mode  Cnt     Score    Error  Units
H3Benchmark.h3Boundary  thrpt   25  1385.246 ± 14.643  ops/s
H3Benchmark.pointToH3   thrpt   25   194.886 ±  4.175  ops/s
```

after
```
Benchmark                Mode  Cnt     Score    Error  Units
H3Benchmark.h3Boundary  thrpt   25  1451.318 ± 21.722  ops/s
H3Benchmark.pointToH3   thrpt   25   205.115 ±  3.841  ops/s
```

### CODE REVIEW

File: libs/h3/src/main/java/org/elasticsearch/h3/FastMath.java [hunk 1]
```diff
@@ -551,9 +560,9 @@ public static double atan2(double y, double x) {
             }
             if (x == Double.NEGATIVE_INFINITY) {
                 if (y == Double.POSITIVE_INFINITY) {
-                    return 3 * Math.PI / 4;
+                    return 3 * Math.PI * 0.25;
```
Comment (craigtaverner) [inline libs/h3/src/main/java/org/elasticsearch/h3/FastMath.java:563]: Could be M_QUARTER_PI here too.

File: libs/h3/src/main/java/org/elasticsearch/h3/FastMath.java [hunk 2]
```diff
@@ -551,9 +560,9 @@ public static double atan2(double y, double x) {
...
                 } else if (y == Double.NEGATIVE_INFINITY) {
-                    return -3 * Math.PI / 4;
+                    return -3 * Math.PI * 0.25;
```
Comment (craigtaverner) [inline libs/h3/src/main/java/org/elasticsearch/h3/FastMath.java:565]: Could be M_QUARTER_PI here too.

**Review by craigtaverner — APPROVED** [review]
Nice improvement!

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**